caput/misc/profiling.py

```diff
@@ -0,0 +1,137 @@
+"""Profiling for the caput pipeline"""
+import collections
+import logging
+import time
+
+import numpy as np
+import psutil
+
+logger = logging.getLogger(__name__)
+
+
+class PSUtilProfiler(psutil.Process):
```

Comment:
Your code seems to treat this as singleton, i.e. you have one master instance and it's keeping track of the different threads of profiling. Is there a reason not to make things easy and just create a new instance every time you want to track things?

Reply:
A few more thoughts about this. I wrote a much more limited version of this for getting the amount of IO done in my branch `caput.misc.IOUsage`, and I think yours will basically be able to replace it, but there's maybe a few nice things it could do to fit in better in my code:
- Allow it to run as a context manager
- Accept a logger object rather than just using the module level logger

Reply:
Also, another thought. You wrap the results in a namedtuple, I wonder if it would be better to do it with a dict. I find it useful to iterate over the items without knowing what they are, and you can't do that with a namedtuple without using semi-private attributes (either `._as_dict()` or `._fields`)

Reply:
Anyway, this all looks great! Thanks.